File: esbmc.py/Testes02/validador_categoria/validador_repetidos_consecutivos.cpp
#define MAX_LEN 31
#define MAX_LEN 31


bool validate_consecutive_repeated(const char category[MAX_LEN], int len) {
    for (int i = 0; i < len - 1; i++) {
        __ESBMC_assert(category[i] != category[i + 1], "A categoria não deve conter caracteres repetidos consecutivos.");
    }
    return true;
}

int main() {
    char category[MAX_LEN];
    int len;

    __ESBMC_assume(len >= 3 && len <= 30);

    validate_consecutive_repeated(category, len);

    return 0;
}

File: esbmc.py/Testes02/validador_categoria/validador_repetidos_consecutivos.py
MAX_LEN: int = 31

def validate_consecutive_repeated(category: str, length: int) -> bool:
    i: int = 0
    while i < length - 1:
        assert category[i] != category[i + 1], \
            "A categoria não deve conter caracteres repetidos consecutivos."
        i += 1
    return True


def main() -> None:
    category: str = input()  
    length: int = len(category)

    assert length >= 3 and length <= 30

    validate_consecutive_repeated(category, length)


main()
